systemtest/approvals/TestAgentConfig.approved.json

```diff
@@ -3,24 +3,17 @@
         {
             "@timestamp": "dynamic",
             "agent_config_applied": 1,
-            "data_stream": {
-                "dataset": "apm.internal",
-                "namespace": "default",
-                "type": "metrics"
-            },
-            "ecs": {
-                "version": "dynamic"
-            },
+            "data_stream.dataset": "apm.internal",
+            "data_stream.namespace": "default",
+            "data_stream.type": "metrics",
```

Comment:
Why did the keys change here? This does not look related to EA v2 changes.

Reply:
This is related to disabling synthetic source. Synthetic source always produces hierarchical objects, but the original docs use flattened fields in some cases.